### ISSUE TITLE
The value values obtained are inconsistent

### ISSUE THREAD

**Issue by hzy4854** [akheron/jansson#464]
```
	std::string data_str1 = "{\"bool\":true,\"int1\":1345,\"number\":13343.234334}";

	json_error_t json_error1;

	auto js1 = json_loads(data_str1.c_str(), data_str1.length(), &json_error1);

	if (js1 == NULL)

	{

		printf("json_error1:%s", json_error1.text);

	}

	else

	{

		const json_t *i1 = json_object_get(js1, "int1");

		printf("int1 is %d\n", json_integer_value(i1));

		printf("int1 is %f\n", json_number_value(i1));

		json_delete(js1);

	}



	std::string data_str2 = "{\"bool\":true,\"int2\":1345,\"number\":13343.234334,\"str\":\"hello,world\"}";

	json_error_t json_error2;

	auto js2 = json_loads(data_str2.c_str(), data_str2.length(), &json_error2);

	if (js1 == NULL)

	{

		printf("json_error2:%s", json_error1.text);

	}

	else

	{

		const json_t *i2 = json_object_get(js2, "int2");

		printf("int2 is %d\n", json_integer_value(i2));

		printf("int2 is %f\n", json_number_value(i2));

		json_delete(js2);

	}
```
result ： 
int1 is 0
int1 is 1345.000000
int2 is 1345
int2 is 1345.000000

**Comment by ploxiln** [comment]
The second argument to `json_loads()` is not length, it is for options flags. You're passing pseudo-random options flags. Corrected:

```c++
auto js1 = json_loads(data_str1.c_str(), 0, &json_error1);
```

**Comment by akheron** [comment]
You're passing the string length as decoding flags to json_loads. Didn't check, but I guess in the first case the flags contain JSON_DECODE_INT_AS_REAL, so int1 is a real number and json_integer_value returns 0.

TL;DR Try passing 0 as the second argument to json_loads.

On a related note, you should use json_decref instead of json_delete. Check the docs.

**Comment by hzy4854** [comment]
thank you
